torch/_dynamo/resume_execution.py

```diff
@@ -31,6 +31,76 @@ class ReenterWith:
     stack_index: int = None
     target_values: Optional[Tuple] = None
 
+    # If we do not want to destroy the stack, we can do the same thing as a
+    # `SETUP_WITH` block, only that we store the context manager in a local_symbol
+    def try_except(self, code_options, cleanup: List[Instruction]):
+        load_args = []
+        if self.target_values:
+            load_args = [
+                create_instruction(
+                    "LOAD_CONST",
+                    PyCodegen.get_const_index(code_options, val),
+                    val,
+                )
+                for val in self.target_values
+            ]
+        ctx_name = f"___context_manager_{self.stack_index}"
+        if ctx_name not in code_options["co_varnames"]:
+            code_options["co_varnames"] += (ctx_name,)
+
+        except_jump_target = create_instruction("NOP")
+        cleanup_complete_jump_target = create_instruction("NOP")
+
+        setup_finally = [
+            *load_args,
+            create_instruction("CALL_FUNCTION", len(load_args)),
+            create_instruction(
+                "STORE_FAST", code_options["co_varnames"].index(ctx_name), ctx_name
+            ),
+            create_instruction(
+                "LOAD_FAST", code_options["co_varnames"].index(ctx_name), ctx_name
+            ),
+            create_instruction("LOAD_METHOD", "__enter__"),
+            create_instruction("CALL_METHOD", 0),
+            create_instruction("POP_TOP"),
+            create_instruction("SETUP_FINALLY", target=except_jump_target),
+        ]
+
+        reset = [
+            create_instruction(
+                "LOAD_FAST", code_options["co_varnames"].index(ctx_name), ctx_name
+            ),
+            create_instruction("LOAD_METHOD", "__exit__"),
+            create_instruction(
+                "LOAD_CONST", PyCodegen.get_const_index(code_options, None), None
+            ),
+            create_instruction("DUP_TOP"),
+            create_instruction("DUP_TOP"),
+            create_instruction("CALL_METHOD", 3),
+            create_instruction("POP_TOP"),
+        ]
+        if sys.version_info < (3, 9):
+            epilogue = [
+                create_instruction("POP_BLOCK"),
+                PyCodegen.create_begin_finally(),
+                except_jump_target,
+                *reset,
+                create_instruction("END_FINALLY"),
+            ]
+        else:
```

Comment:
Let's add 3.11 support here.

Reply:
Seems to have been unnecessary. Cool

Reply:
I have a bunch of 3.11 changes affecting this part that I'm currently trying to merge.